### ISSUE TITLE
[java] NonCaseLabelInSwitch should consider switch expressions

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#5257]
**Affects PMD Version:** 7.6.0

**Rule:** [NonCaseLabelInSwitchStatement](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#noncaselabelinswitchstatement)

**Description:**

It currently only considers switch statements, but switch expressions could also have non case labels.

**Code Sample demonstrating the issue:**

```java
public class Foo {
    void bar(int x) {
        x = switch (x) {
            case 2: int y=8;
            somelabel:  // violation expected
                yield y;
            default:
                int j=8;
                yield j;
        };
    }
}
```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.
